feature/settings/src/main/java/com/google/jetpackcamera/settings/SettingEnabledState.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.jetpackcamera.settings
+
+import com.google.jetpackcamera.settings.model.AspectRatio
+import com.google.jetpackcamera.settings.model.CaptureMode
+import com.google.jetpackcamera.settings.model.DarkMode
+import com.google.jetpackcamera.settings.model.FlashMode
+import com.google.jetpackcamera.settings.model.Stabilization
+import com.google.jetpackcamera.settings.ui.DEVICE_UNSUPPORTED_TAG
+import com.google.jetpackcamera.settings.ui.FPS_UNSUPPORTED_TAG
+import com.google.jetpackcamera.settings.ui.LENS_UNSUPPORTED_TAG
+import com.google.jetpackcamera.settings.ui.STABILIZATION_UNSUPPORTED_TAG
+
+sealed interface SingleSelectableState {
+    data object Selectable : SingleSelectableState
+    data class Disabled(val disabledRationale: Set<DisabledRationale>) : SingleSelectableState {
```

Comment:
Why use a `Set<DisabledRationale>` here instead of just `DisabledRationale`?

Reply:
there could be multiple factors that prevent an option from being enabled

for example, lets say 60 fps is supported by the rear, but not the front lens on a device.
if stabilization is on the front camera, the 60fps would be disabled because of the stabilization mode and the selected lens.
I figured leaving all the constraints it as a set would leave flexibility to display on the UI or debug.

I also wasn't entirely sure how i should prioritize which single DisabledRationale to display when there are multiple

Reply:
Just keep in mind this is UI state; it shouldn't necessarily be making the decision about which is the most important to show, the ViewModel should.

I can see the need to possibly want to show multiple reasons, but it's also possible that those reasons could be collated into a single reason in the ViewModel. So for instance instead of:
```
Disabled(
    setOf(
        DisabledRationale("The lens doesn't support 60 fps"),
        DisabledRationale("The current stabilization mode does not allow 60 fps")
    )
)
```

It could be:
```
Disabled(
    DisabledRationale("The lens and stabilization mode do not allow for 60fps")
)
```

Or, the ViewModel can just decide to show one of the DisabledRationale, and then when that rationale is no longer relevant, it can update the `Disabled` state to show the next DisabledRationale.

Reply:
Which option are you referring to by "the latter"?

My suggestion is to remove the setOf. If you have multiple disabled rationales, just use the first one. If that disabled rationale goes away, then the SettingsUiState will be updated with the next disabled rationale.

Currently you're always using the first disabled rationale in the set, but if there is a reason in the future to use multiple, you can add an overload that takes a set.

Reply:
woops, by latter I was referring to the first snippet 
```
Disabled(
    setOf(
        DisabledRationale("The lens doesn't support 60 fps"),
        DisabledRationale("The current stabilization mode does not allow 60 fps")
    )
)
```
It makes sense to just add an overload down the line should it be necessary